lib/sycamore/sycamore/utils/cache.py

```diff
@@ -16,10 +18,24 @@ def set(self, hash_key: str, hash_value):
         pass
 
     @staticmethod
-    def get_hash_key(data: bytes) -> str:
+    def get_hash_bytes(data: bytes) -> hashlib.sha256:
         hash_sha256 = hashlib.sha256()
         hash_sha256.update(data)
-        return hash_sha256.hexdigest()
+        return hash_sha256
+
+    @staticmethod
+    def get_hash_file(file_path: str) -> hashlib.sha256:
+        sha = hashlib.sha256()
+        with open(file_path, "rb") as file:
+            file_buffer = file.read(BLOCK_SIZE)
+            while len(file_buffer) > 0:
```

Comment:
Can just say: while file_buffer:

I don't like to repeat code, so I usually do:
```
while True:
  file_buffer = file.read(BLOCK_SIZE)
  if not file_buffer:
    break
  sha.update(file_buffer)
```